From 179afdb6606e98f5ff3f7958a548bb08188fd9e2 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Wed, 2 Aug 2017 17:06:41 +0800
Subject: [PATCH] ARM: dts: rockchip: rk3288 add efuse_id for cpuinfo

Add efuse_id for cpuinfo to get system serial number.

Change-Id: If197c2961611364a2cb94972c33171bea105c61b
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-android.dtsi | 6 ++++++
 arch/arm/boot/dts/rk3288.dtsi         | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-android.dtsi b/arch/arm/boot/dts/rk3288-android.dtsi
index 13793d71528f..c3f1239095c6 100644
--- a/arch/arm/boot/dts/rk3288-android.dtsi
+++ b/arch/arm/boot/dts/rk3288-android.dtsi
@@ -50,6 +50,12 @@
 		bootargs = "earlycon=uart8250,mmio32,0xff690000 vmalloc=496M";
 	};
 
+	cpuinfo {
+		compatible = "rockchip,cpuinfo";
+		nvmem-cells = <&efuse_id>;
+		nvmem-cell-names = "id";
+	};
+
 	/delete-node/ dmc@ff610000;
 
 	dfi: dfi {
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index c72cf0799da1..e49268c477c0 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -1618,6 +1618,9 @@
 		clocks = <&cru PCLK_EFUSE256>;
 		clock-names = "pclk_efuse";
 
+		efuse_id: id@7 {
+			reg = <0x7 0x10>;
+		};
 		cpu_leakage: cpu_leakage@17 {
 			reg = <0x17 0x1>;
 		};
-- 
2.35.3

